fpdf/fpdf.py

```diff
@@ -2250,15 +2250,18 @@ def set_font_size(self, size):
         if isclose(self.font_size_pt, size):
             return
         self.font_size_pt = size
-        if self.page > 0:
-            if not self.current_font:
-                raise FPDFException(
-                    "Cannot set font size: a font must be selected first"
-                )
+        self.font_size_set_for_page = False
+
+    def set_font_size_for_page(self):
```

Comment:
```suggestion
    def _set_font_size_for_page(self):
```
Let's use a private method: no need for end-users to fiddle with that neither